simulator/core/simulator.h

```diff
@@ -58,9 +108,41 @@ extern int usetestflash;
 // Simulator state
 extern int cycle;
 #ifdef HAVE_REPLAY
+/** Seek to the specified simulated cycle, if possible.
+ *
+ * This function allows to set the simulator state to a cycle in the past or,
+ * if the cycle state has already been rewound, in the future. It updates the
+ * simulator (and the pipeline if enabled) cycle to the value specified. This
+ * operation can fail if the simulator and pipeline states fall out of sync. If
+ * the target cycle is too far foward, it sets the cycle to the latest possible
+ * cycle the simulator knows.
+ *
+ * @param[in] target Target cycle to seek to.
+ *
+ * @returns True on success, false if it was not able to.
+ * @post On success, the simulator was set to the specified target cycle. On
+ * failure, most likely the pipeline and simulator are no longer in sync.
+ */
 bool simulator_state_seek(int target);
 #endif
+
+// FIXME misnomer
+/** Record the time when the simulator is no longer running (this doesn't
+ * actually check that it's not running, nor does it force the simulator to
+ * stop!).
+ *
+ * The value recorded is used to calculate the effective frequency of the
+ * simulator. This function is not meant to be called when the simulator is
+ * already "sleeping".
+ */
 void sim_sleep(void);
+
+//FIXME misnomer
+/** Reset the recorded start time of the simulator.
```

Comment:
Again a bit of a poorly named variable, it's not restarting the start time per se, it's really marking the time when the simulator started executing again, which is slightly different I think.